<a href="https://colab.research.google.com/github/Kethellem/cursopython/blob/main/AnaliseDesconto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files
uploaded = files.upload()

Saving supermercado.csv to supermercado.csv


In [9]:
import pandas as pd
import plotly.express as px

df = pd.read_csv('supermercado.csv')

print(df.head())

print(df.columns)

                                             title       Marca  Preco_Normal  \
0             Pack 12 un, Leche extra proteína 1 L  Loncoleche         19788   
1  Pack 12 un, Leche chocolate receta original 1 L     Soprole         18228   
2   Pack 12 un, Leche semidescremada chocolate 1 L     Soprole         18228   
3    Pack 12 un, Leche semidescremada frutilla 1 L     Soprole         18228   
4      Pack 12 un, Leche sin lactosa chocolate 1 L  Loncoleche         17988   

   Preco_Desconto  Preco_Anterior  Desconto Categoria  
0               0               0         0   lacteos  
1               0               0         0   lacteos  
2               0               0         0   lacteos  
3               0               0         0   lacteos  
4               0               0         0   lacteos  
Index(['title', 'Marca', 'Preco_Normal', 'Preco_Desconto', 'Preco_Anterior',
       'Desconto', 'Categoria'],
      dtype='object')


In [5]:
# Ajuste o nome da coluna aqui:
coluna = 'Desconto'  # Substitua pelo nome correto

#  Calculando estatísticas
media = df[coluna].mean()
mediana = df[coluna].median()
moda = df[coluna].mode()[0]
desvio_padrao = df[coluna].std()
maximo = df[coluna].max()
minimo = df[coluna].min()

#  Exibindo os resultados
print(f"Média: {media:.2f}")
print(f"Mediana: {mediana:.2f}")
print(f"Moda: {moda:.2f}")
print(f"Desvio Padrão: {desvio_padrao:.2f}")
print(f"Máximo: {maximo}")
print(f"Mínimo: {minimo}")


Média: 68.11
Mediana: 0.00
Moda: 0.00
Desvio Padrão: 241.16
Máximo: 4349
Mínimo: 0


In [14]:
import plotly.express as px

# 🎨 Histograma
fig = px.histogram(
    df,
    x=coluna,
    nbins=30,  # Número de barras
    title=f'Distribuição de {coluna}',
    color_discrete_sequence=['indianred']
)

fig.update_layout(
    xaxis_title=coluna,
    yaxis_title='Frequência',
    title_font_size=20,
    bargap=0.1,
    template='plotly_white'
)
fig.show()

# Boxplot
fig2 = px.box(
    df,
    y=coluna,
    title=f'Boxplot de {coluna}',
    color_discrete_sequence=['mediumseagreen']
)

fig2.update_layout(
    yaxis_title=coluna,
    title_font_size=20,
    template='plotly_white'
)
fig2.show()



Analise das vendas

A média dos descontos foi 68.11.
A mediana indica 0.00.
Visualizando a distribuição, percebemos que o desconto max foi 4349
